reviews/tests.py

```diff
@@ -267,3 +300,168 @@ def test_auto_review_task(self):
 
         reasons = auto_review_task(self.study, s1_t1)
         self.assertEqual(len(reasons), 1)
+
+
+class BaseViewTestCase(BaseReviewTestCase):
```

Comment:
The reason is that `TestCase` does not follow the Mixin design pattern. You could easily end up calling a non-existent `super().setUp()` because `setUp()` isn't defined inthe base class.

That's probably solvable with a stub base class, but I´d rather do the generalizing work when we need it so we actually know what problems need solving. I've renamed the class to `BaseReviewViewTestCase` to better reflect that it's more of a bespoke thing for reviews only.

Reply:
> I've renamed the class to BaseReviewViewTestCase to better reflect that it's more of a bespoke thing for reviews only. (For now, I totally support using this more in the future!)

Good enough for now ;) 

However, some of your argumentation for why a mixin wouldn't work doesn't hold up in my eyes:

> The reason is that ``TestCase`` does not follow the Mixin design pattern.

So? The fact that Django's ``TestCase`` doesn't use Mixins doesn't mean it's not a valid design pattern to use here. It only means Django didn't need it. [Django's own tests](https://github.com/django/django/blob/78da5ca0c1f2ab3201f8f6cd629e80d805ea023d/tests/test_utils/test_serializemixin.py) do use them ;)

> You could easily end up calling a non-existent ``super().setUp()`` because ``setUp()`` isn't defined inthe base class.

I don't see how. ``setUp()`` is defined on [topmost base class](https://docs.python.org/3/library/unittest.html#unittest.TestCase) of Django's ``TestCase``. 

And you're never going to use a Mixin intended for ``TestCase`` outside of a ``TestCase``; just like you would never use a ``View`` mixin outside of a ``View``. (Well, you obviously can, but shouldn't).

Reply:
I can't recall the problem I ran into earlier, but the BaseViewTestCase and all other tests now follow the mixin pattern. The other tests had to be adjusted accordingly.

Reply:
A more general location would maybe be DSC? The mixin is general enough to use in other apps as well.

Otherwise, the `main` app is (in my eyes) a good place for app-wide code. Lots of view mixins for example already live there